### ISSUE TITLE
Unable to stop server on mac

### ISSUE THREAD

**Issue by lyw07** [learningequality/kolibri#6307]
### Observed behavior

When running on mac, I started to randomly get logging messages
```
INFO     ENGINE Caught signal SIGTERM.
INFO     ENGINE Bus STOPPING
```
Sometimes, the program is able to continue running; sometimes, it isn't. 
But every time after this error shows up, I am not able to stop the server using Ctrl+C.
The program hangs at this step for more than 10 minutes before I use Ctrl+Z to kill it:
```
INFO     ENGINE Waiting for child threads to terminate...
```
This does not happen when I ran Kolibri on Ubuntu both foreground and background.

### Expected behavior

Users should be able to stop the server successfully

### User-facing consequences

Users will have to kill the program and kill all the corresponding python processes to stop Kolibri

### Errors and logs

```
INFO     Starting Kolibri 0.13.0b7
INFO     ENGINE Listening for SIGTERM.
INFO     ENGINE Listening for SIGHUP.
INFO     ENGINE Listening for SIGUSR1.
INFO     ENGINE Listening for SIGINT.
INFO     ENGINE Bus STARTING
INFO     Started new JOBCHECKER thread ID#42378ffb12594e50807bde3603407cc5
INFO     Started new JOBCHECKER thread ID#f4471e9694eb47e0873a6b77aedd77e2
INFO     Started new SCHEDULECHECKER thread ID#ed80030c77d949daa0c31b5b87b6b977
INFO     Registering ourselves to zeroconf network with id '74aefcce9d7550b3021615db97b7b484'...
INFO     ENGINE Serving on http://0.0.0.0:8080
INFO     ENGINE Bus STARTED
INFO     Ping succeeded! (response: {'id': 506061, 'messages': []})
INFO     Sqlite database Vacuum finished.
INFO     ENGINE Caught signal SIGTERM.
INFO     ENGINE Bus STOPPING
INFO     ENGINE Caught signal SIGTERM.
INFO     ENGINE Caught signal SIGTERM.
INFO     ENGINE Bus STOPPING
INFO     ENGINE Bus STOPPING
INFO     ENGINE Caught signal SIGTERM.
INFO     ENGINE Caught signal SIGTERM.
INFO     ENGINE Bus STOPPING
INFO     ENGINE Bus STOPPING
INFO     ENGINE Caught signal SIGTERM.
INFO     ENGINE Bus STOPPING
INFO     ENGINE Caught signal SIGTERM.
INFO     ENGINE Bus STOPPING
INFO     ENGINE Caught signal SIGTERM.
INFO     ENGINE Caught signal SIGTERM.
INFO     ENGINE Caught signal SIGTERM.
INFO     ENGINE Bus STOPPING
INFO     ENGINE Bus STOPPING
INFO     ENGINE Bus STOPPING
INFO     ENGINE Caught signal SIGTERM.
INFO     ENGINE Bus STOPPING
^C  <--- trying to stop the server with Ctrl+C here
INFO     ENGINE Caught signal SIGINT.
INFO     ENGINE Bus STOPPING
INFO     ENGINE Caught signal SIGINT.
INFO     ENGINE Bus STOPPING
INFO     ENGINE Caught signal SIGINT.
INFO     ENGINE Caught signal SIGINT.
INFO     ENGINE Caught signal SIGINT.
INFO     ENGINE Bus STOPPING
INFO     ENGINE Bus STOPPING
INFO     ENGINE Caught signal SIGINT.
INFO     ENGINE Caught signal SIGINT.
INFO     ENGINE Caught signal SIGINT.
INFO     ENGINE Caught signal SIGINT.
INFO     ENGINE Bus STOPPING
INFO     ENGINE Bus STOPPING
INFO     ENGINE Bus STOPPING
INFO     ENGINE Caught signal SIGINT.
INFO     ENGINE Bus STOPPING
INFO     ENGINE Bus STOPPING
INFO     ENGINE Bus STOPPING
INFO     ENGINE HTTP Server cherrypy._cpwsgi_server.CPWSGIServer(('0.0.0.0', 8080)) shut down
INFO     
Kolibri instance '74aefcce9d7550b3021615db97b7b484' has left the zeroconf network.

WARNING  JOBCHECKER shut down event received; closing.
WARNING  SCHEDULECHECKER shut down event received; closing.
WARNING  JOBCHECKER shut down event received; closing.
INFO     ENGINE Bus STOPPED
INFO     ENGINE Bus EXITING
INFO     ENGINE Bus EXITED
INFO     ENGINE Waiting for child threads to terminate...
^Z <--- kill the program with Ctrl+Z here after waiting for 10 minutes
```

### Steps to reproduce

1. on mac, start kolibri
2. after the setup wizard, check the log, if it shows `ENGINE Caught signal SIGTERM.`, try to stop the server
3. if it doesn't show the error message, then import a channel until it shows up.

### Context

Kolibri version: 0.13.0b7 (both python2 and 3)
Operating system: MacOSX
Browser: Google Chrome

**Comment by rtibbles** [comment]
Hrm - it seems like the async task terminating is triggering the `SIGTERM` handler in the parent process.

**Comment by indirectlylit** [comment]
@lyw07 this is the beta7 pex?

**Comment by lyw07** [comment]
@indirectlylit no, it's beta7 whl

**Comment by rtibbles** [comment]
Fixed in #6360
